src/content/docs/zh-cn/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,598 @@
+
+```astro del={3} ins={4} title="src/components/MyAstroComponent.astro"
+<style>
+  div {
+   color: var(--my-value);
+   color: var(--myValue);
+  }
+</style>
+```
+
+### 移除：自动展平 `getStaticPaths()` 的返回值
+
+在 Astro v2.x 中，`getStaticPaths()` 的返回值会自动展平，以允许你返回一个数组的数组而不会出错。
+
+Astro v3.0 移除了对 `getStaticPaths()` 返回值的自动展平。
+
+#### 我应该怎么做？
+
+如果你 return 一个数组的数组而不是一个数组的 _对象_（这是预期的），`.flatMap` 和 `.flat` 现在应该被用来确保你返回一个平的数组。
+
+如果你需要更新代码，Astro 将抛出一个错误消息，指出 `getStaticPath()` 的返回值必须是一个对象数组。
+
+### 移动：`astro check` 现在需要一个外部包
+
+在 Astro v2.x 中，默认情况下包含了 [`astro check`](/zh-cn/reference/cli-reference/#astro-check)，并且它的依赖项被打包在 Astro 中。这意味着无论你是否使用 `astro check`，都会有一个更大的包。这也阻止你对 TypeScript 和 Astro 语言服务器的版本进行控制。
+
+Astro v3.0 将 `astro check` 命令从 Astro 核心中移出，并且现在需要一个外部包 `@astrojs/check`。此外，你必须在你的项目中安装 `typescript` 来使用 `astro check` 命令。
+
+#### 我应该怎么做？
+
+在升级 Astro v3.0 后运行 `astro check` 命令，并按照提示安装所需的依赖项，或者手动将 `@astrojs/check` 和 `typescript` 安装到你的项目中。
+
+### 废弃：`build.excludeMiddleware` 和 `build.split`
+
+在 Astro v2.x 中，`build.excludeMiddleware` 和 `build.split` 用于在 SSR 模式下使用适配器时更改特定文件的产出方式。
+
+Astro v3.0 用新的 [SSR 适配器配置选项](/zh-cn/guides/integrations-guide/#官方集成) 替换了这些构建配置选项，以执行相同的任务：`edgeMiddleware` 和 `functionPerRoute`。
+
+#### 我应该怎么做？
+
+更新 Astro 配置文件，直接在适配器配置中使用新的选项。
+
+```js title="astro.config.mjs" del={5-7} ins={9}
+import { defineConfig } from "astro/config";
+import vercel from "@astrojs/vercel/serverless";
+
+export default defineConfig({
+    build: {
+      excludeMiddleware: true
+    },
+    adapter: vercel({
+      edgeMiddleware: true
+    }),
+});
+```
+
+```js title="astro.config.mjs" del={5-7} ins={9}
+import { defineConfig } from "astro/config";
+import netlify from "@astrojs/netlify/functions";
+
+export default defineConfig({
+     build: {
+        split: true
+     },
+     adapter: netlify({
+        functionPerRoute: true
+     }),
+});
+```
+
+### 废弃：`markdown.drafts`
+
+在 Astro v2.x 中，`markdown.drafts` 配置允许你在运行开发服务器时拥有草稿页面，但在生产环境中不会构建。
+
+Astro v3.0 废弃了这个功能，而是使用内容集合的方法来处理草稿页面，通过手动过滤来实现，这样可以更好地控制这个功能。
+
+#### 我应该怎么做？
+
+为了继续将你项目中的某些页面标记为草稿，请 [迁移到内容集合](/zh-cn/guides/content-collections/#从基于文件的路由迁移)，并使用 `draft: true` frontmatter 属性 [手动过滤页面](/zh-cn/guides/content-collections/#筛选集合查询)。
+
+### 默认值改变：端口 `3000`
+
+在 Astro v2.x 中，默认情况下，Astro 运行在 `3000` 端口上。
+
+Astro v3.0 将 [默认端口](/zh-cn/reference/cli-reference/#--port-number) 更改为 `4321`。
+
+#### 我应该怎么做？
+
+更新任何现有的引用 `localhost:3000`，例如在测试或在你的 `README` 中，以反映新的端口 `localhost:4321`。
+
+### 默认值改变：import.meta.env.BASE_URL `trailingSlash`
+
+在 Astro v2.x 中，`import.meta.env.BASE_URL` 默认情况下会将你的 [`base`](/zh-cn/reference/configuration-reference/#base) 设置附加到 [trailingSlash](/zh-cn/reference/configuration-reference/#trailingslash)。`trailingSlash: "ignore"` 也会附加一个尾随斜杠。
+
+Astro v3.0 不再默认情况下将 `import.meta.env.BASE_URL` 与尾随斜杠附加在一起，也不会在设置 `trailingSlash: "ignore"` 时附加一个尾随斜杠。（`base` 与 `trailingSlash: "always"` 或 `trailingSlash: "never"` 结合使用的现有行为保持不变。）
```

Comment:
Could use some rewording. Waiting for confirmation of whether or not my understanding of the original English docs is correct before suggesting changes.